shared/src/commonMain/kotlin/io/customer/shared/device/UserAgentStore.kt

```diff
@@ -0,0 +1,18 @@
+package io.customer.shared.device
+
+/**
+ * Interface to make it easy for wrapper SDKs to provide user agent lazily to KMM.
+ */
+interface UserAgentStore {
```

Comment:
By adding this, we can keep SDK responsible for building user agents and still use it in KMM.

Reply:
Ideally, we would want to move this logic into the `Networking Client` module, since that module is part of the KMM library we should move this there. But since the last PR review. we saw some feedback not to move all components at once, this interface will act as a bridge to keep login in native SDKs and be used in KMM.

Reply:
Yes, correct. Also, user agent relies on a lot of data e.g. device info, OS, etc. and I don't really want to move the responsibility to generate data to KMM in phase 1 other than directly related to BQ.